### DISCUSSION TITLE
Progress circle stop after loading album cover

### DISCUSSION THREAD

**Discussion by Argo16x** [rern/rAudio#1084] [Bug Reports]
Hello!

I'm using rAudio as an upnp renderer with BubblePNP.
With latest versions the song progress circle and time stop immediately after loading thr album coverart (if rAudio finds it). The problem is not present on version 20220904 (which I'm using now).

Kind regards,
Simone

**Comment by markblinkhorn** [comment]
I installled 20260401 today and found that this problem is still present. It hasn't worked for the last few releases and I don't remember when it stopped; I've just been ignoring it for a while. Anyhow, I had a quick dig around today and I've found a workaround. I have no idea if this will break anything else - maybe @rern could comment? 


I edited the javascript file at /srv/http/assets/js/function.js (around line 1514) to change:

```javascript
                V.interval.elapsed = setInterval( () => {
                        S.elapsed++;
                        if ( ! S.Time || S.elapsed < S.Time ) {
                                if ( V.localhost ) {
                                        PROGRESS.arc( S.elapsed / S.Time );
                                        $( '#time-bar' ).css( 'width', S.elapsed / S.Time * 100 +'%' );
                                }
                                elapsedhms = COMMON.second2HMS( S.elapsed );
                                $elapsed.text( elapsedhms );
                                if ( S.state !== 'play' ) UTIL.intervalClear( 'elapsed' );
                        } else {
                                S.elapsed = 0;
                                UTIL.intervalClear();
                                $elapsed.empty();
                                PROGRESS.set( 0 );
                        }
                }, 1000 );
 
``` 

to:

```javascript
                V.interval.elapsed = setInterval( () => {
                        S.elapsed++;
                        if ( ! S.Time || S.elapsed < S.Time ) {
                                PROGRESS.arc( S.elapsed / S.Time );
                                if ( V.localhost ) {
                                        $( '#time-bar' ).css( 'width', S.elapsed / S.Time * 100 +'%' );
                                }
                                elapsedhms = COMMON.second2HMS( S.elapsed );
                                $elapsed.text( elapsedhms );
                                if ( S.state !== 'play' ) UTIL.intervalClear( 'elapsed' );
                        } else {
                                S.elapsed = 0;
                                UTIL.intervalClear();
                                $elapsed.empty();
                                PROGRESS.set( 0 );
                        }
                }, 1000 );
 
``` 

As you can see, I have moved the **`PROGRESS.arc( S.elapsed / S.Time );`** statement to the outside of the `if` statement. The progress arc functionality is restored and I haven't found anything else not working (yet!)

**Comment by markblinkhorn** [comment]
Oh, I forgot to add; you need to restart rAudio and refresh your browser to make it work.

**Comment by rern** [comment]
Please try the UPDATE.

**Reply by markblinkhorn** [reply]
Confirmed. PROGRESS.arc works.

Correction - it has a problem with refreshing the webpage view - for example, when swapping tabs in the browser. The arc starts from the beginning if you swap back to the rAudio tab while a track is playing and, if it is a long track, hitting f5 causes the arc to fill the circle. I've gone back to the 'main' version, which seems to behave more as expected.
